ERCS/erc-7717.md

```diff
@@ -0,0 +1,250 @@
+     */
+    function getExpiration(address signer) external view returns (uint256);
+
+    /**
+     * @notice Checks if a given address is a valid signer as of the current time.
+     * @param signer The address to check.
+     * @return True if the address is a valid signer, false otherwise.
+     */
+    function isValidSigner(address signer) external view returns (bool);
+
+    /**
+     * @notice Verifies if a given hash and signature pair is valid. It uses EIP-712 typed data signing.
+     * @param hash The hash of the typed data.
+     * @param signature The signature to verify.
+     * @return magicValue The EIP-1271 magic value if the signature is valid, 0x0 otherwise.
+     */
+    function isValidSignature(bytes32 hash, bytes memory signature) external view returns (bytes4 magicValue);
+
+    /**
+     * @notice Verifies if a given message and signature pair is valid. It's an alternative function for legacy message signing.
+     * @param message The message that was signed.
+     * @param signature The signature to verify.
+     * @return magicValue A specific legacy magic value if the signature is valid, 0x0 otherwise.
+     */
+    function isValidSignature(bytes memory message, bytes memory signature) external view returns (bytes4 magicValue);
+}
+
+```
+
+#### Ownership transfer function
+
+When computing the address of the SCA underlying the deterministic account, the `DeterministicAccountFactory` sets itself as the owner. So when the deterministic account is actually deployed, `DeterministicAccountFactory` goes through a two-step process where it first sets itself as the owner and then transfer the ownership to a new address provided by the user.
+
+Since the two steps need to be handled atomically, `DeterministicAccountFactory` must be provided with the SCA ownership transfer function when deploying the account. Unfortunately, multiple SCA implementations employ different function names for transferring ownership, so it is necessary to add the calldata for each SCA implementation that the service provider wishes to support.
+
+Our reference implementation includes transfer ownership calldata for Alchemy’s Light/Modular accounts and ZeroDev’s Kernel v2.4.
+
+## Rationale
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+- Key Management
+
+A service provider should rotate its `securer` key regularly. Since the `securer` address must not change after it has been used to compute the address of a deterministic account, the service must not blindly discard the `securer` key after a certain time period. It should first check that there are no deterministic accounts that are secured by the `securer` key it will discard, and if there are any such accounts, the service provider should either postpone discarding it or keep a valid signature on its server for future use.
+
+## Reference Implementation
+
+See `https://github.com/deterministic-aa/monorepo`
```

Comment:
Please remove this external link. You may place a reference implementation inline (if it's CC0), or in the assets directory (if it's large or under another non-copyleft license.)